### PR TITLE
supplemental: make private class final with default constructor

### PR DESCRIPTION
**Pull request by Kevin222004**
supplemental: make private class final with default constructor

part of :- https://github.com/checkstyle/checkstyle/pull/12737

### CODE REVIEW

**Comment by Kevin222004** [comment]
@romani please merge soon if possible

**Comment by romani** [comment]
Please reference issue in tail of commit message.

@rnveach , we are changing API classes, should we do something to notify users?

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClassClassWithPrivateCtorWithNestedExtendingClass.java [hunk 1]
```diff
@@ -10,17 +10,18 @@
 public class InputFinalClassClassWithPrivateCtorWithNestedExtendingClass {
     class A {
         private A() {}
-        private class ExtendA extends A {}
+        private final class ExtendA extends A {}
```
Comment (rnveach) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClassClassWithPrivateCtorWithNestedExtendingClass.java:13]: Inputs should not be modified.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClassClassWithPrivateCtorWithNestedExtendingClass.java (hunk 1 above)
Comment (Kevin222004) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClassClassWithPrivateCtorWithNestedExtendingClass.java:13]: @rnveach that no issue i will remove it. after our new changes that is expected

**Comment by rnveach** [comment]
Only way to notify users is by issue label. If we consider this a breaking change, it will need its own issue.

However, all of the classes are `private static`. They are technically internal classes, and if they were "extensible", we would be getting compile errors. Users outside of Checkstyle can't access a `private static` class without reflection, and I don't think you can extend with reflection.

> in this case why we need final on them

Because we are following the check which is active and will violate this going forward.

If you disagree with changes to check, then please discuss in the issue.
If you wish to not continue using this check, then please create a new issue.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClassClassWithPrivateCtorWithNestedExtendingClass.java (hunk 1 above)
Comment (Kevin222004) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClassClassWithPrivateCtorWithNestedExtendingClass.java]: done

**Comment by Kevin222004** [comment]
travis is not because of changes

**Comment by Kevin222004** [comment]
@romani @rnveach please merge this so this https://github.com/checkstyle/checkstyle/pull/12737 you find easy to review

**Review by rnveach — APPROVED** [review]
I am fine to merge this without any special issue since all classes are private.

**Review by romani — APPROVED** [review]

**Pull request merged**